vsphere/data_source_vsphere_resource_pool_test.go

```diff
@@ -0,0 +1,127 @@
+package vsphere
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceVSphereResourcePool(t *testing.T) {
+	var tp *testing.T
+	testAccDataSourceVSphereResourcePoolCases := []struct {
+		name     string
+		testCase resource.TestCase
+	}{
+		{
+			"basic",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccDataSourceVSphereResourcePoolPreCheck(tp)
+				},
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config: testAccDataSourceVSphereResourcePoolConfig(),
+						Check: resource.ComposeTestCheckFunc(
+							resource.TestMatchResourceAttr("data.vsphere_resource_pool.pool", "id", regexp.MustCompile("^resgroup-")),
+						),
+					},
+				},
+			},
+		},
+		{
+			"no datacenter and absolute path",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccDataSourceVSphereResourcePoolPreCheck(tp)
+				},
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config: testAccDataSourceVSphereResourcePoolConfigAbsolutePath(),
+						Check: resource.ComposeTestCheckFunc(
+							resource.TestMatchResourceAttr("data.vsphere_resource_pool.pool", "id", regexp.MustCompile("^resgroup-")),
+						),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testAccDataSourceVSphereResourcePoolCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tp = t
+			resource.Test(t, tc.testCase)
+		})
+	}
+}
+
+func testAccDataSourceVSphereResourcePoolPreCheck(t *testing.T) {
+	if os.Getenv("VSPHERE_DATACENTER") == "" {
+		t.Skip("set VSPHERE_DATACENTER to run vsphere_resource_pool data source acceptance tests")
+	}
+	if os.Getenv("VSPHERE_CLUSTER") == "" {
+		t.Skip("set VSPHERE_CLUSTER to run vsphere_resource_pool data source acceptance tests")
+	}
+	if os.Getenv("VSPHERE_RESOURCE_POOL") == "" {
+		t.Skip("set VSPHERE_RESOURCE_POOL to run vsphere_resource_pool data source acceptance tests")
+	}
+}
+
+func testAccDataSourceVSphereResourcePoolConfig() string {
+	return fmt.Sprintf(`
+variable "datacenter" {
+  default = "%s"
+}
+
+variable "cluster" {
+  default = "%s"
+}
+
+variable "resource_pool" {
+  default = "%s"
+}
+
+data "vsphere_datacenter" "dc" {
+  name = "${var.datacenter}"
+}
+
+data "vsphere_resource_pool" "pool" {
+  name          = "${var.cluster}/Resources/${var.resource_pool}"
```

Comment:
Like most things in gomomi's [`Finder`](https://godoc.org/github.com/vmware/govmomi/find#Finder), this is actually a path search. `name` can be `pool`, it can be `cluster/Resources/pool`, it could even be `cluster/Resources/pool/subpool`, or `/dc1/host/cluster/Resources/pool/subpool`. `datacenter_id` is allowable because Finder has a concept of setting a datacenter to search off of, but it doesn't have a concept of cluster or compute resource.

This could possibly work if Finder had a way to do sub-path wildcards, but it doesn't look like it does. For example, if we allowed the cluster to be specified (which would need to be searched on as well), and combining the path in a way like /dc/host/cluster/.../subpool (subbing `...` for `**` or some other type of standard glob) worked, then this might be possible, but it isn't at this point in time.

I think a couple of approaches here might be workable:

* Have `name` also match the name of a cluster, and then just use the default resource pool from there.
* Have a `cluster` path similar to above (and maybe name it `default_for_cluster`, although I don't know if that's too linguistic) and make the two attributes conflict. I would probably prefer this approach. Then, we can just slap `Resources` to the end of the path and just search off of that. The idea here is that someone should be using it if they want the top-level resource pool for a cluster and otherwise use `name` explicitly.

Reply:
For now, and probably for v1.0.0, I want to keep this as it is in the spirit of moving this along. I've added #250 for this as a TODO.